hudi-common/src/main/java/org/apache/hudi/common/util/PartitionPathEncodeUtils.java

```diff
@@ -71,7 +73,7 @@ public static String escapePathName(String path, String defaultPath) {
       if (defaultPath == null) {
         //previously, when path is empty or null and no default path is specified,
         // __HIVE_DEFAULT_PARTITION__ was the return value for escapePathName
-        return "__HIVE_DEFAULT_PARTITION__";
+        return HUDI_DEFAULT_PARTITION_PATH;
```

Comment:
why is this change needed

Reply:
+1

Reply:
When the partition's value is null, hudi use "default" as the part of path, rather than "__HIVE_DEFAULT_PARTITION__".
And I review the implement when need to encode partition path before #2645,  like `getRecordPartitionPath` and `getPartitionPath` in KeyGenUtils. I think this just encode path, and use "default" when partitionPath is null or empty.
So i guess maybe this is forgotten to change.